crates/signs/src/bars.rs

```diff
@@ -182,6 +188,15 @@ impl Material for BarMaterial {
 #[derive(Component)]
 struct BarChild(Entity);
 
+#[derive(Component)]
+struct BarVisibilityTimer(Timer);
+
+impl Default for BarVisibilityTimer {
+    fn default() -> Self {
+        Self(Timer::from_seconds(3., TimerMode::Once))
```

Comment:
I would use a constant `UPDATE_VISIBILITY_DURATION` (defined near the top of the file) and change this to `Timer::new(UPDATE_VISIBILITY_DURATION, TimerMode::Once)`.